spi/core-spi/src/main/java/org/eclipse/dataspaceconnector/spi/query/QueryResolver.java

```diff
@@ -0,0 +1,33 @@
+/*
+ *  Copyright (c) 2020 - 2022 Microsoft Corporation
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Microsoft Corporation - initial API and implementation
+ *
+ */
+package org.eclipse.dataspaceconnector.spi.query;
+
+import java.util.stream.Stream;
+
+/**
+ * Class responsible for applying queries on an input elements of type T. Currently supports querying a stream of elements.
+ *
+ * @param <T> type of the queried elements.
+ */
+public abstract class QueryResolver<T> {
```

Comment:
could this be an interface, even a `@FunctionalInterface`?

Reply:
fully agree, changed